### PR TITLE
ibus: add auto-start management interface to ibus-setup

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

**Comment by pyhalov** [comment]
It's work-in-progress attempt to add "enable/disable on startup" management to ibus-setup properties. The question is  - do we need it if imf-selector can do it for us?

File: components/inputmethod/ibus/patches/01-enable-autostart-management.patch [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+-            os.symlink(ibus_desktop, link_file)
++        return False
++
++    def __checkbutton_auto_start_toggled_cb(self, button):
++        IBUS_SCRIPT="/etc/X11/xinit/imf.d/0010.ibus"
++
++        home = os.getenv("HOME")
++        if home:
++            if self.__checkbutton_auto_start.get_active():
++               os.system("/usr/bin/cat %s > %s/.imf" % (IBUS_SCRIPT, home))
```
Comment (xen0l) [inline components/inputmethod/ibus/patches/01-enable-autostart-management.patch]: Why os.system and not subprocess.call or subprocess.check_call?

Comment (pyhalov) [reply components/inputmethod/ibus/patches/01-enable-autostart-management.patch]: What's the actual difference?  We need to execute shell command. I'd worried more about path escaping here, perhaps, this exact call could be rewritten in clear python.

Comment (pyhalov) [reply components/inputmethod/ibus/patches/01-enable-autostart-management.patch]: Avoided using os.system here. Also new code shouldn't care about file path escaping

**Comment by kenmays** [comment]
LGTM

File: components/inputmethod/ibus/patches/01-enable-autostart-management.patch [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
++        try:
++            home = os.getenv("HOME")
++            if home:
++                config_file = path.join(home, ".imf")
++                if self.__checkbutton_auto_start.get_active():
++                   shutil.copyfile(IBUS_SCRIPT, config_file)
++                elif os.path.exists(config_file):
++                   os.unlink(config_file)
++        except Exception as e:
++            dlg = Gtk.MessageDialog(message_type = Gtk.MessageType.ERROR,
```
Comment (xen0l) [inline components/inputmethod/ibus/patches/01-enable-autostart-management.patch:66]:
```
dlg = Gtk.MessageDialog(message_type = Gtk.MessageType.ERROR,
```
->
```
dlg = Gtk.MessageDialog(message_type=Gtk.MessageType.ERROR,
```

etc

**Review by xen0l — CHANGES_REQUESTED** [review]

File: components/inputmethod/ibus/patches/01-enable-autostart-management.patch (hunk 2 above)
Comment (pyhalov) [reply components/inputmethod/ibus/patches/01-enable-autostart-management.patch:66]: It's a common style in that file

**Pull request merged**